src/useFish.ts

```diff
@@ -103,188 +72,26 @@ export type ReactFish<Command, PublicState> = {
-                return stream$.map<PublicState, ReactFish<Command, PublicState>>(state => ({
-                  state,
-                  feed: (command: Command) => feed(entityFish, name)(command).toPromise(),
-                  name,
-                  stream$
-                }))
-              })
-            )
-      ),
-      share()
-    )
-
-    const sub = allStates$.subscribe(s => setState([s, allStates$.map(s => s.map(st => st.state))]))
-    return () => sub.unsubscribe()
-  }, [])
-
-  return state
-}
-
-/**
- * Observe the registry fish and map the state to an array of fish names, and finally observe all entity fish according to
- * the fish names and return a array of ReactFish representing the entities.
- *
- * Additionally you will get at Observable of the stream
- *
- * @see useRegistryFish If your registry fish has an array of FishNames or strings as public state
- *
- * @param registryFish registry fish to map the state to the entity fish.
- * @param mapFn mapper from RegistryPublicState to a array of the entity fish names
- * @param entityFish fish, represent a single entity in the registry fish
- * @typeParam RegistryPublicState observable state of the registry fish. Used in the mapper function
- * @typeParam Command Available commands for the fish
- * @typeParam PublicState Observable state of the fish
- * @returns [ReactFish[], Observable<PublicState[]>] returns a ReactFish array and a Observable with all states as array @see useStream
- */
-export const useRegistryFishMap = <RegistryPublicState, Command, PublicState>(
-  registryFish: FishTypeRegistryMap<RegistryPublicState>,
-  mapFn: (regState: RegistryPublicState) => string[],
-  entityFish: FishTypeEntity<Command, PublicState>
-): UseRegistryFishReturn<Command, PublicState> => {
-  const { observe, feed } = usePond()
-  const [state, setState] = React.useState<UseRegistryFishReturn<Command, PublicState>>([
-    [],
-    Observable.never<ReadonlyArray<PublicState>>().startWith([])
-  ])
-
-  React.useEffect(() => {
-    const allStates$ = observe(registryFish, 'reg').pipe(
-      map((state: RegistryPublicState) => mapFn(state)),
-      switchMap(reg =>
-        reg.length === 0
-          ? Observable.never<ReadonlyArray<ReactFish<Command, PublicState>>>().startWith([])
-          : combineLatest(
-              reg.map(name => {
-                const stream$ = observe(entityFish, name).share()
-                return stream$.map<PublicState, ReactFish<Command, PublicState>>(state => ({
-                  state,
-                  feed: (command: Command) => feed(entityFish, name)(command).toPromise(),
-                  name,
-                  stream$
-                }))
-              })
-            )
-      ),
-      share()
-    )
+export const useFish = <State, E>(fish: Fish<State, E>): ReactFish<State, E> => {
+  const pond = usePond()
+  const mkReactFish = (state: State): ReactFish<State, E> => ({
+    state,
+    run: x => pond.run(fish, x).toPromise()
```

Comment:
perhaps move the creation of this closure out of the function to save some allocations in case V8 is not as intelligent as we’d like it to be